pro/src/ui-kit/form/SelectAutoComplete2/SelectAutocomplete.tsx

```diff
@@ -0,0 +1,310 @@
+  pluralLabel?: string
+  smallLabel?: boolean
+}
+
+const SelectAutocomplete = ({
+  className,
+  disabled = false,
+  fieldName,
+  hideArrow,
+  hideFooter = false,
+  hideTags = false,
+  inline,
+  isOptional = false,
+  label,
+  maxHeight,
+  multi = false,
+  options,
+  placeholder,
+  pluralLabel,
+  smallLabel = false,
+}: SelectAutocompleteProps): JSX.Element => {
+  const { setFieldTouched, setFieldValue } = useFormikContext<any>()
+
+  const [field, meta] = useField<string | string[]>(fieldName)
+  const [searchField, searchMeta] = useField(`search-${fieldName}`)
+
+  const [hovered, setHovered] = useState<number | null>(null)
+  const [optionsLabelById, setOptionsLabelById] = useState<
+    Record<string, string>
+  >({})
+
+  const containerRef = useRef<HTMLDivElement>(null)
+  const listRef = useRef<HTMLUListElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+  const [filteredOptions, setFilteredOptions] = useState(options)
+
+  useEffect(() => {
+    setFilteredOptions(options)
+  }, [options])
+
+  useEffect(() => {
+    if (isOpen && searchField.value !== '')
+      setFieldValue(`search-${fieldName}`, '', false)
+  }, [isOpen])
+
+  /* hashtable for the options */
+  useEffect(() => {
+    setOptionsLabelById(
+      options.reduce<Record<string, string>>((optionsById, option) => {
+        optionsById[option.value] = option.label
+        return optionsById
+      }, {})
+    )
+  }, [options])
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent): void => {
+      if (!containerRef.current?.contains(e.target as Node)) {
+        setIsOpen(false)
+      }
+    }
+    if (containerRef.current) {
+      document.addEventListener('mousedown', handleClickOutside)
+    }
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [containerRef])
+
+  useEffect(() => {
+    const regExp = new RegExp(searchField.value, 'i')
+    setFilteredOptions(
+      options.filter(
+        option => searchField.value === '' || option.label.match(regExp)
+      )
+    )
+    setHovered(null)
+  }, [searchField.value])
+
+  /* istanbul ignore next */
+  const handleKeyDown: KeyboardEventHandler<HTMLDivElement> = event => {
+    switch (event.key) {
+      case 'ArrowUp':
+        if (hovered === null) {
+          setHovered(filteredOptions.length - 1)
+        } else if (hovered <= 0) {
+          setHovered(null)
+        } else {
+          setHovered(hovered - 1)
+        }
+        if (!isOpen) setIsOpen(true)
+        listRef.current?.focus()
+        break
+      case 'ArrowDown':
+        if (hovered === null) {
+          setHovered(0)
+        } else if (hovered >= filteredOptions.length - 1) {
+          setHovered(null)
+        } else {
+          setHovered(hovered + 1)
+        }
+        if (!isOpen) setIsOpen(true)
+        listRef.current?.focus()
+        break
+      case 'Space':
+        if (!isOpen) setIsOpen(true)
```

Comment:
(avec la ligne d'après)
```suggestion        
        openField()
```